File: rxanimation/src/main/java/com/reyurnible/rxanimation/animator/RxAnimator.java
package com.reyurnible.rxanimation.animator;

import android.animation.Animator;
import android.animation.ValueAnimator;
import android.annotation.TargetApi;
import android.os.Build;
import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;

import rx.Observable;

/**
 * Static factory methods for animator event observable.
 */
public final class RxAnimator {

    @CheckResult
    @NonNull
    public static Observable<AnimatorEvent> events(@NonNull Animator animation) {
        return Observable.create(new AnimatorEventOnSubscribe(animation));
    }

    @TargetApi(Build.VERSION_CODES.KITKAT)
    @CheckResult
    @NonNull
    public static Observable<AnimatorPauseEvent> pauseEvents(@NonNull Animator animation) {
        return Observable.create(new AnimatorPauseEventOnSubscribe(animation));
    }

    @CheckResult
    @NonNull
    public static Observable<AnimatorUpdateEvent> updateEvents(@NonNull ValueAnimator animation) {
        return Observable.create(new AnimatorUpdateEventOnSubscribe(animation));
    }

}

File: rxanimation-kotlin/src/main/kotlin/com/reyurnible/rxanimation_kotlin/animation/RxAnimator.kt
package com.reyurnible.rxanimation_kotlin.animation

import android.animation.Animator
import android.animation.ValueAnimator
import com.reyurnible.rxanimation.animator.AnimatorEvent
import com.reyurnible.rxanimation.animator.AnimatorPauseEvent
import com.reyurnible.rxanimation.animator.AnimatorUpdateEvent
import com.reyurnible.rxanimation.animator.RxAnimator
import rx.Observable

fun Animator.events(): Observable<AnimatorEvent> = RxAnimator.events(this)

fun Animator.pauseEvents(): Observable<AnimatorPauseEvent> = RxAnimator.pauseEvents(this)

fun ValueAnimator.updateEvents(): Observable<AnimatorUpdateEvent> = RxAnimator.updateEvents(this)
